libshaderc/include/shaderc.h

```diff
@@ -141,6 +141,46 @@ void shaderc_compile_options_set_disassembly_mode(
 void shaderc_compile_options_set_forced_version_profile(
     shaderc_compile_options_t options, int version, shaderc_profile profile);
 
+// To support file inclusion, libshaderc invokes a callback into its client to
+// resolve the full path and content of the included file.
+// Client callback should follow the specified function signature below, and
+// it should be passed to libshaderc through the corresponding setter function.
+// When the including of a file is done, libshaderc will call another client
+// callback to clean up the resources used for the including process.  Client
+// should implement the clean up method and pass it to libshaderc together with
+// the response method.
+
+// The struct that contains the information to be returned to the includer.
+// The client-side implemented response method should return this struct. And
+// the underlying data is owned by client code.
+struct shaderc_includer_response {
+  const char* path;
+  size_t path_length;
+  const char* content;
+  size_t content_length;
+};
+
+// The function signature of the client-side implemented reponse method. It
+// returns shaderc_includer_reponsestruct.
+typedef shaderc_includer_response* (
+    *shaderc_includer_response_get_fn)(void* user_data,
```

Comment:
Wait a second.  This interface is missing information.
It should take two things:
-   const char * context_path that gives the full path of the file that is doing the including.   
- the filename being requested.

That's required to support stuff like this:
  first file includes   "subdir/a.h"
  and then subdir/a.h  includes "b.h"
For normal processing, the includer should try to find "b.h" as "subdir/b.h".  But to know it must do that, it must know which file is doing the including. This is the whole reason we wanted the path element in the includer response in the first place.

We also have to spell out what happens on the *first* inclusion.  We could make the rule that context_path is either nullptr or an empty string.  (Empty string would play nicer in the C++ space where the context path would probably be a const std::string& parameter.)  So probably empty string is better.